Xamarin.PropertyEditing.Mac/Controls/Custom/NumericSpinEditor.cs

```diff
@@ -0,0 +1,287 @@
+			var controlSize = NSControlSize.Small;
+
+			stepper = new NSStepper {
+				TranslatesAutoresizingMaskIntoConstraints = false,
+				ValueWraps = false,
+			};
+			stepper.Cell.ControlSize = controlSize;
+
+			formatter = new NSNumberFormatter {
+				FormatterBehavior = NSNumberFormatterBehavior.Version_10_4,
+				Locale = NSLocale.CurrentLocale,
+				MaximumFractionDigits = 15,
+				NumberStyle = NSNumberFormatterStyle.Decimal,
+				UsesGroupingSeparator = false,
+			};
+
+			numericEditor = new NumericTextField {
+				Alignment = NSTextAlignment.Right,
+				Formatter = formatter,
+				TranslatesAutoresizingMaskIntoConstraints = false,
+			};
+
+			numericEditor.Cell.ControlSize = controlSize;
+			numericEditor.Cell.Font = NSFont.ControlContentFontOfSize (NSFont.SystemFontSizeForControlSize (controlSize));
+
+			stepper.Activated += (s, e) => {
+				OnStepperActivated (s, e);
+			};
+
+			numericEditor.ValidatedEditingEnded += (s, e) => {
+				OnFieldEditingEnded (s, e);
+			};
+
+			AddSubview (stepper);
+			AddSubview (numericEditor);
+
+			this.DoConstraints (new[] {
+				numericEditor.ConstraintTo (this, (n, c) => n.Width == c.Width - 17),
+				stepper.ConstraintTo (numericEditor, (s, n) => s.Left == n.Right + 5),
+				stepper.ConstraintTo (numericEditor, (s, n) => s.Top == n.Top + 1),
+			});
+		}
+
+		protected virtual void SetStepperActivated ()
+		{
+			SetValue (stepper.DoubleValue);
+		}
+
+		protected void OnStepperActivated (object sender, EventArgs e)
+		{
+			if (!editing) {
+				editing = true;
+				SetStepperActivated ();
+				OnStepperValueChanged ();
+				if (ValueChanged != null)
+					ValueChanged (this, EventArgs.Empty);
+				editing = false;
+			}
+		}
+
+		protected virtual void OnStepperValueChanged ()
+		{
+			if (StepperValueChanged != null)
+				StepperValueChanged (this, EventArgs.Empty);
+		}
+
+		protected void OnFieldEditingEnded (object sender, EventArgs e)
+		{
+			if (!editing) {
+				editing = true;
+				SetValue (numericEditor.StringValue);
+				if (FieldEditingEnded != null)
+					FieldEditingEnded (this, EventArgs.Empty);
+				if (ValueChanged != null)
+					ValueChanged (this, EventArgs.Empty);
+				editing = false;
+			}
+		}
+
+		public virtual void SetValue (string value)
+		{
+			//Regulates maximun and minium out of range
+			stepper.DoubleValue = CoerceValue (FieldValidation.FixInitialValue (value, Value.ToEditorString ()).ToEditorDouble ());
+			numericEditor.StringValue = FieldValidation.RoundDoubleValue (stepper.DoubleValue.ToEditorString (), NumericMode == ValidationType.Decimal ? FieldValidation.DefaultXcodeMaxRoundDigits : 0);
+		}
+
+		public virtual void SetValue (double value)
+		{
+			//Regulates maximun and minium out of range
+			stepper.DoubleValue = CoerceValue (value);
+			numericEditor.StringValue = FieldValidation.RoundDoubleValue (stepper.DoubleValue.ToEditorString (), NumericMode == ValidationType.Decimal ? FieldValidation.DefaultXcodeMaxRoundDigits : 0);
+		}
```

Comment:
I don't like how we have two `SetValue` overrides, both of which are pretty complicated. Any control which overrides one must also override the second, and we have had bugs in the past where that was not done. Can we simplify this? Also, does `CoerceValue` also need to be public and overridable?